src/server/editorServices.ts

```diff
@@ -2764,6 +2768,10 @@ namespace ts.server {
                     this.hostConfiguration.watchOptions = convertWatchOptions(args.watchOptions);
                     this.logger.info(`Host watch options changed to ${JSON.stringify(this.hostConfiguration.watchOptions)}, it will be take effect for next watches.`);
                 }
+
+                if(args.typeAcquisition) {
+                    this.hostConfiguration.typeAcquisition = { ...this.hostConfiguration.typeAcquisition, ...args.typeAcquisition };

```

Comment:
Why spread in the new setting when others (like `extraFileExtensions`) seem to completely replace?

Reply:
Agreed this should be direct assignment instead

Reply:
I was going off of `UserPreferences` above. I guess direct assignment would expect the editor to be sending all properties each time?

Reply:
there arent too many properties so it is better to be single assignment